From c45889334c75933c99de4195c8559a4a7aba8892 Mon Sep 17 00:00:00 2001
From: Antonio de Angelis <Antonio.deAngelis@arm.com>
Date: Tue, 23 Aug 2022 13:06:07 +0100
Subject: [PATCH 9/9] Initialise driver wrappers as first step in
 psa_crypto_init()

This patch amends the order of initialisations performed in psa_crypto_init()
to make sure that the driver wrappers based on the PSA driver API are initialised
before anything else.

Signed-off-by: Antonio de Angelis <antonio.deangelis@arm.com>
---
 library/psa_crypto.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/library/psa_crypto.c b/library/psa_crypto.c
index 8cc5629f9..b0159991c 100644
--- a/library/psa_crypto.c
+++ b/library/psa_crypto.c
@@ -6306,6 +6306,11 @@ psa_status_t psa_crypto_init( void )
     if( global_data.initialized != 0 )
         return( PSA_SUCCESS );
 
+    /* Init drivers */
+    status = psa_driver_wrapper_init( );
+    if( status != PSA_SUCCESS )
+        goto exit;
+
     /* Initialize and seed the random generator. */
     mbedtls_psa_random_init( &global_data.rng );
     global_data.rng_state = RNG_INITIALIZED;
@@ -6318,11 +6323,6 @@ psa_status_t psa_crypto_init( void )
     if( status != PSA_SUCCESS )
         goto exit;
 
-    /* Init drivers */
-    status = psa_driver_wrapper_init( );
-    if( status != PSA_SUCCESS )
-        goto exit;
-
 #if defined(PSA_CRYPTO_STORAGE_HAS_TRANSACTIONS)
     status = psa_crypto_load_transaction( );
     if( status == PSA_SUCCESS )
-- 
2.25.1

